src/wallet/mod.rs

```diff
@@ -389,6 +457,39 @@ where
         Ok(new_addresses_cached)
     }
 
+    /// Return vector of n unused addresses from the [`KeychainKind`].
+    /// If less than n unused addresses are returned, the rest will be populated by new addresses.
+    /// The unused addresses returned are in order of oldest in keychain first, with increasing index.
+    pub fn get_batch_unused_addresses(
+        &self,
+        n: usize,
+        keychain: KeychainKind,
+    ) -> Result<Vec<AddressInfo>, Error> {
+        let unused_key_indexes = self.get_unused_key_indexes(keychain)?;
+        let mut addresses = unused_key_indexes
```

Comment:
So to line up with my other comments this should just iter over script pubkeys from database and check if each one is used successively until you've found `n` of them (and then of course get more if you need).